src/utils/CustomPropTypes.js

```diff
@@ -57,4 +66,18 @@ function createMountableChecker() {
   return createChainableTypeChecker(validate);
 }
 
+function createKeyOfChecker(obj) {
+  function validate(props, propName, componentName) {
+    let propValue = props[propName];
+    if (!obj.hasOwnProperty(propValue)) {
+      let valuesString = JSON.stringify(Object.keys(obj));
+      return new Error(
+        ('Invalid prop `' + propName + '` of value `' + propValue + '` ') +
+        ('supplied to `' + componentName + '`, expected one of ' + valuesString + '.')
```

Comment:
That's fine with me. I think they're a bit odd anyway. The only issue is consistency with React error messages, but that seems minor.